jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/ProcessServiceImplWithoutAuditTest.java

```diff
@@ -0,0 +1,390 @@
+
+    	// get all variables
+    	Map<String, Object> variables = processService.getProcessInstanceVariables(deploymentUnit.getIdentifier(), processInstanceId);
+    	assertNotNull(variables);
+    	assertEquals(2, variables.size());
+    	assertTrue(variables.containsKey("approval_document"));
+    	assertTrue(variables.containsKey("approval_reviewComment"));
+    	assertEquals("test", variables.get("approval_document"));
+    	assertEquals("need review", variables.get("approval_reviewComment"));
+
+    	// now change single variable
+    	processService.setProcessVariable(deploymentUnit.getIdentifier(), processInstanceId, "approval_reviewComment", "updated review comment");
+    	// let's verify it
+    	variableValue = processService.getProcessInstanceVariable(deploymentUnit.getIdentifier(), processInstanceId, "approval_reviewComment");
+    	assertNotNull(variableValue);
+    	assertTrue(variableValue instanceof String);
+    	assertEquals("updated review comment", variableValue);
+
+    	// and lastly let's update both variables
+    	params = new HashMap<String, Object>();
+    	params.put("approval_document", "updated document");
+        params.put("approval_reviewComment", "final review");
+
+        processService.setProcessVariables(deploymentUnit.getIdentifier(), processInstanceId, params);
+        variables = processService.getProcessInstanceVariables(deploymentUnit.getIdentifier(), processInstanceId);
+    	assertNotNull(variables);
+    	assertEquals(2, variables.size());
+    	assertTrue(variables.containsKey("approval_document"));
+    	assertTrue(variables.containsKey("approval_reviewComment"));
+    	assertEquals("updated document", variables.get("approval_document"));
+    	assertEquals("final review", variables.get("approval_reviewComment"));
+
+    	processService.abortProcessInstance(deploymentUnit.getIdentifier(), processInstanceId);
+
+    	pi = processService.getProcessInstance(deploymentUnit.getIdentifier(), processInstanceId);
+    	assertNull(pi);
+    }
+
+    @Test
+    public void testStartProcessAndAbortThenChangeVariables() {
+        assertNotNull(deploymentService);
+
+        KModuleDeploymentUnit deploymentUnit = new KModuleDeploymentUnit(GROUP_ID, ARTIFACT_ID, VERSION);
+        deploymentService.deploy(deploymentUnit);
+        units.add(deploymentUnit);
+
+        Map<String, Object> params = new HashMap<String, Object>();
+        params.put("approval_document", "test");
+        params.put("approval_reviewComment", "need review");
+
+        long processInstanceId = processService.startProcess(deploymentUnit.getIdentifier(), "org.jbpm.writedocument", params);
+        assertNotNull(processInstanceId);
+        
+        processService.abortProcessInstance(deploymentUnit.getIdentifier(), processInstanceId);
+
+        ProcessInstance pi = processService.getProcessInstance(deploymentUnit.getIdentifier(), processInstanceId);
+        assertNull(pi);
+        
+        try {
```

Comment:
This and following try-catch blocks probably repeat; it might be worth making a utility method(s) of them. 

Secondly, one tip - AssertJ offers very nice capabilities for dealing with exceptions http://joel-costigliola.github.io/assertj/assertj-core-features-highlight.html#exception-assertion.

Reply:
applied, using assertj to check exceptions